app/models/trade/annual_report_upload.rb

```diff
@@ -84,8 +84,56 @@ def process_validation_rules
     end
   end
 
-  def submit
-    # TODO
+  def run_primary_validations
+    @validation_errors = run_validations(
+      Trade::ValidationRule.where(:is_primary => true)
+    )
+  end
+
+  def submit(submitter)
+    run_primary_validations
+    unless @validation_errors.count == 0
+      self.errors[:base] << "Submit failed, primary validation errors present."
+      return false
+    end
+
+    return false unless sandbox.copy_from_sandbox_to_shipments(submitter)
+    # remove uploaded file
+    #store_dir = csv_source_file.store_dir
+    #remove_csv_source_file!
+    #puts '### removing uploads dir ###'
+    #puts Rails.root.join('public', store_dir)
+    #FileUtils.remove_dir(Rails.root.join('public', store_dir), :force => true)
+
+    # remove sandbox table
+    sandbox.destroy
+
+    # clear downloads cache
+    #DownloadsCacheCleanupJob.perform_async(:shipments)
+
+    # flag as submitted
+    submitter_type = submitter.class.to_s.split(':').first
+    if submitter_type == 'Epix'
+      update_column(:epix_submitted_at, DateTime.now)
+      update_column(:epix_submitted_by_id, submitter.id)
+    else
+      update_column(:submitted_at, DateTime.now)
+      update_column(:submitted_by_id, submitter.id)
+    end
```

Comment:
The number of records submitted is the integer coming out of the sql function, right?
Looking at the code it seems it can only be equal to the number of rows (also another field already in the model) the aru had originally. The `copy_from_sandbox_to_shipments` method returns true only when all the rows have been moved, because the SQL function returns -1 otherwise.
Is this correct?